medical_pathology_import/wizards/medical_pathology_import.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, fields, models
+
+
+class MedicalPathologyImport(models.TransientModel):
+    """ It provides a model for importing ICD-10-CM codes to Odoo. """
+
+    _name = 'medical.pathology.import'
+    _description = 'Medical Pathology Import'
+
+    zip_uri = fields.Char(
+        help='URI to full ZIP file',
+        string='ZIP File URI',
+    )
+    file_name = fields.Char(
+        help='Name of file (inside of ZIP) representing full tabular XML',
+        string='XML File Name',
+    )
+    code_type_id = fields.Many2one(
+        string='Code Type',
+        comodel_name='medical.pathology.code.type',
+        required=True,
+        help='Identifies the code type that is being imported. If you are not'
+             ' sure what to choose, select an importer first and leave the'
+             ' resulting default',
+    )
+    importer_type = fields.Selection(
+        selection=lambda s: s._get_importer_types(),
+        required=True,
+        string='Importer Type',
+    )
+
+    @api.model
+    def _get_importer_types(self):
+        return []
+
+    @api.multi
+    @api.onchange('importer_type')
+    def _onchange_importer_type(self):
+        """ It calls gets default data by importer type.
+
+        To plug into this interface, importers should implement method
+        ``_onchange_importer_type_*``, such as
+        ``_onchange_importer_type_icd10``.
+        """
+        try:
+            method = getattr(
+                self, '_onchange_importer_type_%s' % self.importer_type,
+            )
+            method()
+        except AttributeError:
+            self.zip_uri = False
+            self.file_name = False
+
+    @api.model
+    def create(self, vals):
+        """ It begins import process after create """
+        record = super(MedicalPathologyImport, self).create(vals)
+        record.do_import()
```

Comment:
Let's add a context key to skip this. `no_import=True`

Never know when you need to skip a long operation when coding.